test/e2e/network/dual_stack.go

```diff
@@ -204,6 +206,165 @@ var _ = SIGDescribe("[Feature:IPv6DualStackAlphaFeature] [LinuxOnly]", func() {
 
 		assertNetworkConnectivity(f, *serverPods, *clientPods, "dualstack-test-client", "80")
 	})
+
+	ginkgo.It("should create service with cluster ip from primary service range [Feature:IPv6DualStackAlphaFeature:Phase2]", func() {
+		serviceName := "defaultclusterip"
+		ns := f.Namespace.Name
+		jig := e2eservice.NewTestJig(cs, serviceName)
+
+		defaultIPFamily := v1.IPv4Protocol
+		if framework.TestContext.ClusterIsIPv6() {
+			defaultIPFamily = v1.IPv6Protocol
+		}
+
+		t := e2eservice.NewServerTest(cs, ns, serviceName)
+		defer func() {
+			defer ginkgo.GinkgoRecover()
+			errs := t.Cleanup()
+			if len(errs) != 0 {
+				framework.Failf("errors in cleanup: %v", errs)
+			}
+		}()
+
+		ginkgo.By("creating service " + serviceName + " with Service.Spec.IPFamily not set" + ns)
+		service := createService(t.ServiceName, t.Namespace, t.Labels, nil)
+
+		jig.Labels = t.Labels
+		jig.CreateServicePods(cs, ns, 2)
+		result, err := t.CreateService(service)
+		framework.ExpectNoError(err, "failed to create service: %s in namespace: %s", serviceName, ns)
+
+		if len(result.Spec.Ports) != 2 {
+			framework.Failf("got unexpected len(Spec.Ports) for new service: %v", result)
+		}
+		// check the spec has been set to default ip family
+		if result.Spec.IPFamily == nil || *result.Spec.IPFamily != defaultIPFamily {
+			framework.Failf("got unexpected service ip family for new service: %v", *result.Spec.IPFamily)
+		}
+		// check to ensure primary service range used for service with no spec defined
+		if !framework.TestContext.ClusterIsIPv6() && !isIPv4(result.Spec.ClusterIP) {
+			framework.Failf("got unexpected service ip %s, primary service ip family %s", result.Spec.ClusterIP, framework.TestContext.IPFamily)
+		}
+		// ensure endpoints belong to same ipfamily as service
+		endpoints, err := cs.CoreV1().Endpoints(service.Namespace).Get(service.Name, metav1.GetOptions{})
```

Comment:
nit: as this is a single endpoint, so endpoints -> endpoint?

Reply:
Done!